### ISSUE TITLE
Custom comparator

### ISSUE THREAD

**Issue by iDmple** [YannBrrd/elasticsearch-entity-resolution#14]
Hi Yann,

I need to use WeightedLevenshtein which is available in Duke.

In Duke I set the parameters like this:

    <object class="no.priv.garshol.duke.comparators.WeightedLevenshtein$DefaultWeightEstimator"
            name="weight">
        <param name="digit-weight" value="3.8"/>
    </object>

then create the comparator named wl like this:

    <object class="no.priv.garshol.duke.comparators.WeightedLevenshtein"
            name="wl">
        <param name="estimator" value="weight"/>
    </object>

and then it is available to be used:

        <property>
            <name>StreetAndNr</name>
            <comparator>wl</comparator>
            <low>0.3</low>
            <high>0.55</high>
        </property>

I could not find how to do this in your documentation.

Could you please provide an example?

Many thanks!

**Comment by YannBrrd** [comment]
Hi,

Did you succeed finally ?

Is it linked to #15 ?

Le mer. 23 sept. 2015 09:53, iDmple <notifications@github.com> a écrit :

> Hi Yann,
>
> I need to use WeightedLevenshtein which is available in Duke.
>
> In Duke I set the parameters like this:
>
> <object class="no.priv.garshol.duke.comparators.WeightedLevenshtein$DefaultWeightEstimator"
>         name="weight">
>     <param name="digit-weight" value="3.8"/>
> </object>
>
> then create the comparator named wl like this:
>
> <object class="no.priv.garshol.duke.comparators.WeightedLevenshtein"
>         name="wl">
>     <param name="estimator" value="weight"/>
> </object>
>
> and then it is available to be used:
>
>     <property>
>         <name>StreetAndNr</name>
>         <comparator>wl</comparator>
>         <low>0.3</low>
>         <high>0.55</high>
>     </property>
>
> I could not find how to do this in your documentation.
>
> Could you please provide an example?
>
> Many thanks!
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/YannBrrd/elasticsearch-entity-resolution/issues/14>.
>
-- 
Cordialement,
Yann Barraud

**Comment by iDmple** [comment]
No, I didn't. I'm not sure what syntax to use because the weights are set in a subclass of the comparator, unlike the Geo comparator you used in your examples.
It isn't linked to #15.

**Comment by YannBrrd** [comment]
ok

Le jeu. 24 sept. 2015 à 09:08, iDmple <notifications@github.com> a écrit :

> No, I didn't. I'm not sure what syntax to use because the weights are set
> in a subclass of the comparator, unlike the Geo comparator you used in your
> examples.
> It isn't linked to #15.
>
>
>
> On Sep 24, 2015, 08:59, at 08:59, Yann Barraud <notifications@github.com>
> wrote:
> >Hi,
> >
> >Did you succeed finally ?
> >
> >Is it linked to #15 ?
> >
> >Le mer. 23 sept. 2015 09:53, iDmple <notifications@github.com> a écrit
> >:
> >
> >> Hi Yann,
> >>
> >> I need to use WeightedLevenshtein which is available in Duke.
> >>
> >> In Duke I set the parameters like this:
> >>
> >> <object
>
> >class="no.priv.garshol.duke.comparators.WeightedLevenshtein$DefaultWeightEstimator"
> >> name="weight">
> >> <param name="digit-weight" value="3.8"/>
> >> </object>
> >>
> >> then create the comparator named wl like this:
> >>
> >> <object class="no.priv.garshol.duke.comparators.WeightedLevenshtein"
> >> name="wl">
> >> <param name="estimator" value="weight"/>
> >> </object>
> >>
> >> and then it is available to be used:
> >>
> >> <property>
> >> <name>StreetAndNr</name>
> >> <comparator>wl</comparator>
> >> <low>0.3</low>
> >> <high>0.55</high>
> >> </property>
> >>
> >> I could not find how to do this in your documentation.
> >>
> >> Could you please provide an example?
> >>
> >> Many thanks!
> >>
> >> —
> >> Reply to this email directly or view it on GitHub
> >>
> ><https://github.com/YannBrrd/elasticsearch-entity-resolution/issues/14>.
> >>
> >--
> >Cordialement,
> >Yann Barraud
> >
> >
> >---
> >Reply to this email directly or view it on GitHub:
> >
> https://github.com/YannBrrd/elasticsearch-entity-resolution/issues/14#issuecomment-142834035
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/YannBrrd/elasticsearch-entity-resolution/issues/14#issuecomment-142835811>
> .
>
-- 
Cordialement,
Yann Barraud

**Comment by NTBTI** [comment]
Hi Yann
I just came across this while searching for ways to use WeightedLevenshtein and am having the same issue as OP. I'm using the most recent version....is there a way to use this with your excellent tool?

**Comment by iDmple** [comment]
Hi, I still need this functionality.

I also made a lot of bug fixes to the original code. I need to find time for a pull request. Tell me if it's urgent.

Best regards,
Nathalie

**Comment by AbsoluteResults** [comment]
Sorry iDmple....are you saying you have WeightedLevenshtein working and just looking to do a PR? If you have it committed I would love to try it out and see how you implemented it. If not I hope Yann can give some guidance as we'd really like to implement this feature for an upcoming release.

**Comment by iDmple** [comment]
No, I don't. However I have many other bug fixes. I should really review everything and PR them.

Cheers

**Comment by AbsoluteResults** [comment]
I forked the project and made very simple changes so WeightedLevenshtein can be called using all the Duke options (digit-weight, punctuation-weight, etc). Should be accessible through my profile. If a pull request is desired I am more than happy to do that.

**Comment by YannBrrd** [comment]
Please feel free to do so !

Le mar. 26 avr. 2016 19:40, AbsoluteResults <notifications@github.com> a
écrit :

> I forked the project and made very simple changes so WeightedLevenshtein
> can be called using all the Duke options (digit-weight, punctuation-weight,
> etc). Should be accessible through my profile. If a pull request is desired
> I am more than happy to do that.
>
> —
> You are receiving this because you commented.
>
> Reply to this email directly or view it on GitHub
> <https://github.com/YannBrrd/elasticsearch-entity-resolution/issues/14#issuecomment-214823526>
>
-- 
Cordialement,
Yann Barraud

**Comment by AbsoluteResults** [comment]
Done and done! #28 thanks so much
